doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -835,27 +802,72 @@ Inductive definitions
 
 We gradually explain simple inductive types, simple annotated inductive
 types, simple parametric inductive types, mutually inductive types. We
-explain also co-inductive types.
+also explain co-inductive types.
 
 Simple inductive types
 ~~~~~~~~~~~~~~~~~~~~~~
 
-.. cmd:: Inductive @ident : {? @sort } := {? | } @ident : @type {* | @ident : @type }
+.. cmd:: Inductive @inductive_definition {* with @inductive_definition }
+
+   .. insertprodn inductive_definition record_binder_body
+
+   .. prodn::
+      inductive_definition ::= {? > } @ident_decl {* @binder } {? : @term } {? := {? @constructor_list_or_record_decl } } {? @decl_notations }
+      constructor_list_or_record_decl ::= %| {+| @constructor }
```

Comment:
> Also, the leading slash should be optional.

Huh?? "slash" is "/", right?

Reply:
Yeah. I meant pipe.

Reply:
The pipe is unconditional in the code.  fullGrammar has:

```
constructor_list_or_record_decl: [
| "|" LIST1 constructor SEP "|"
   : 
```
Perhaps you would submit an issue for this?

Reply:
The full list of rules is:

```
constructor_list_or_record_decl: [
| "|" LIST1 constructor SEP "|"
| identref constructor_type "|" LIST0 constructor SEP "|"
| identref constructor_type
| identref "{" record_fields "}"
| "{" record_fields "}"
|
]
```

The second and third rules show that the leading pipe is actually optional.

Reply:
I'm currently fixing the issue with the trailing pipe. The reason why this simple rule is split in three and sometimes `constructor` is inlined and sometimes it is not is to disambiguate the case where there is a leading `identref` as it could be followed by a record or a `constructor_type`.